src/coreComponents/mesh/generators/MeshGeneratorBase.hpp

```diff
@@ -19,19 +19,42 @@
 #ifndef GEOSX_MESH_GENERATORS_MESHGENERATORBASE_HPP
 #define GEOSX_MESH_GENERATORS_MESHGENERATORBASE_HPP
 
+//#include "mesh/MeshBody.hpp"
 #include "dataRepository/Group.hpp"
 #include "dataRepository/WrapperBase.hpp"
 #include "codingUtilities/Utilities.hpp"
 #include "common/DataTypes.hpp"
 
+#include "mesh/mpiCommunications/SpatialPartition.hpp"
+#include "mesh/generators/CellBlockManager.hpp"
+
 namespace geosx
 {
 
 namespace dataRepository
 {}
 
+class MeshBody;
+
 class DomainPartition;
 
+class MeshGeneratorHelper
+{
+public:
+  bool hasMetisNeighborList() const { return m_hasMetisNeighborList; }
+  void setHasMetisNeighborList( bool value ) { m_hasMetisNeighborList = value; }
+  std::set<int> & getMetisNeighborList() {return m_metisNeighborList; }
```

Comment:
Returning a reference to non-const is basically not very different from having public attributes.
Once you get the reference, you cannot intercept any usage of the attribute which escaped your watch anymore.

Reply:
Done